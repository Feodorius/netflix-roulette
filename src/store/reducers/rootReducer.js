import initialState from "../initialState";
import { ACTION_FILTER, ACTION_SORT, ACTION_SEARCH, ACTION_GET_MOVIES } from "../../utils/constants";

const rootReducer = (state = initialState, { type, payload }) => {
    switch (type) {
        case ACTION_FILTER: {
            return { ...state, filterOption: payload };
        }
        case ACTION_SORT: {
            return { ...state, sortOption: payload };
        }
        case ACTION_SEARCH: {
            return { ...state, searchString: payload };
        }
        case ACTION_GET_MOVIES: {
            const { movies, totalMovies } = payload;               
            return { ...state, movies, totalMovies };
        }
        default: {
            return state;
        }
    }
};

export default rootReducer;