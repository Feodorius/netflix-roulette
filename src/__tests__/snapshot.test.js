import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore } from "redux";
import rootReducer from "../store/reducers/rootReducer";
import { BrowserRouter as Router } from "react-router-dom";

import FilterBar from '../components/filter-bar/filter-bar.component';
import Search from "../components/search/search.component";
import AddEditDialog from "../components/add-edit-dialog/add-edit-dialog.component";

describe('Should render components properly:', () => {
    const store = createStore(rootReducer);

    test('render filter bar', () => {
        const { asFragment } = render(<Provider store={store}><Router><FilterBar /></Router></Provider>);
        expect(asFragment(<FilterBar />)).toMatchSnapshot();
    });
    test('render search panel', () => {
        const { asFragment } = render(<Provider store={store}><Router><Search /></Router></Provider>);
        expect(asFragment(<Search />)).toMatchSnapshot();
    });
    test('render Add/Edit Dialog', () => {
        const { asFragment } = render(<Provider store={store}><Router><AddEditDialog /></Router></Provider>);
        expect(asFragment(<AddEditDialog />)).toMatchSnapshot();
    });
});