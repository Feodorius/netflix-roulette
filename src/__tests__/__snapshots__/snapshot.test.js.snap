// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render components properly: render Add/Edit Dialog 1`] = `<DocumentFragment />`;

exports[`Should render components properly: render add button 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root makeStyles-contained-3 MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root add-button css-1c5feuk-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    + ADD MOVIE
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Should render components properly: render filter bar 1`] = `
<DocumentFragment>
  <div
    class="MuiTabs-root css-1ujnqem-MuiTabs-root"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
      style="overflow: hidden; margin-bottom: 0px;"
    >
      <div
        class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
        role="tablist"
      >
        <button
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="0"
          type="button"
        >
          all
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          documentary
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          comedy
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          horror
        </button>
        <button
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary filter-tab css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
          role="tab"
          tabindex="-1"
          type="button"
        >
          crime
        </button>
      </div>
      <span
        class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
        style="left: 0px; width: 0px;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should render components properly: render search panel 1`] = `
<DocumentFragment>
  <div
    class="header"
  >
    <div
      class="header-background"
    />
    <div
      class="logo"
    >
      <span
        class="bold"
      >
        netflix
      </span>
      <span
        class="regular"
      >
        roulette
      </span>
    </div>
    <button
      class="MuiButton-root makeStyles-contained-1 MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root add-button css-1c5feuk-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      + ADD MOVIE
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="header-center"
    >
      <h1
        class="heading"
      >
        FIND YOUR MOVIE
      </h1>
      <div
        class="search"
      >
        <div
          class="MuiFormControl-root MuiTextField-root search-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              placeholder="What do you want to watch?"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1br1jja"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root makeStyles-contained-2 MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
          height="56px"
          tabindex="0"
          type="button"
          width="232px"
        >
          SEARCH
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
